SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -90,46 +152,72 @@ function Set-Secret
         [Parameter(ValueFromPipelineByPropertyName)]
         [hashtable] $AdditionalParameters
     )
-    if($Secret -is [string]) {
-        $Secret = @{
-            URL = "http://sn"
-            Notes = $Secret
-        }
-    }
-
     $sb = [System.Text.StringBuilder]::new()
-    if($Secret.UserName) {
-        $sb.Append("Username: ").AppendLine($Secret.UserName)
-        $sb.Append("login: ").AppendLine($Secret.UserName)
+    
+    
+    if ($Secret -is [string]) {
+        $Secret = @{URL = ' http://sn';Notes = $Secret}
+    } elseif ($Secret -is [pscredential]) {
+        $Secret = @{Username = $Secret.Username; Password = $Secret.GetNetworkCredential().password}
     }
 
-    if($Secret.Password) {
-        $pass = $Secret.Password
-        if ($Secret -is [pscredential]) {
-            $pass = $Secret.GetNetworkCredential().password
-        } elseif ($pass -is [securestring]) {
-            $pass = $pass | ConvertFrom-SecureString
-        }
 
-        $sb.Append("Password: ").AppendLine($pass)
-        $sb.Append("password: ").AppendLine($pass)
-        $sb.Append("sudo_password: ").AppendLine($pass)
-    }
+    
+    if ($Secret -is [hashtable]){
+        $SpecialKeys = @('Language', 'NoteType', 'Notes')
 
-    if($Secret.URL) {
-        $sb.Append("URL: ").AppendLine($Secret.URL)
-    }
+        if ($Secret.Keys.count -eq 1 -and $null -ne $Secret.Notes) {
+            $Secret.URL = 'http://sn'
+        }
 
-    if($Secret.Notes) {
-        $sb.AppendLine("Notes:").AppendLine($Secret.Notes)
-    }
+        $Keys = $Secret.Keys.Where({$_ -notin $SpecialKeys })
+        foreach ($k in $Keys) {
+            if ($Secret.$k -is [securestring]) {
+                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.$k)
+                $Secret.$k = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+            }
+            [Void]($sb.AppendLine("$($k): $($Secret.$k)"))
```

Comment:
```suggestion
            $sb.AppendLine("$($k): $($Secret.$k)") | Out-Null
```

Out-Null is fine here. It's the fastest in PS7.

Reply:
Nice to know.